en/components/general/wpf_to_angular_guide/structural_directives.md

```diff
@@ -0,0 +1,146 @@
+
+### `ngIf` Directive
+Now, let’s explore each of these directives, starting with the `ngIf`. This directive allows us to show or hide elements based on a boolean condition. We will start by creating a "div" element with an "h2" tag containing a name.
+```html
+<div>
+    <h2>John</h2>
+</div>
+```
+If we save this our browser will render the name John. However, let’s say we have some type of boolean expression that we want to base the condition of the visibility of this "h2" tag on. For example, we are going to add a property called "isFirstName" and set it to false. In order to tell our div to be rendered when isFirstName equals true, we should use the following syntax *ngIf = "isFirstName".
+```typescript
+  public isFirstName = false;
+```
+``` html
+<div *ngIf="isFirstName">
+  <h2>John</h2>
+</div>
+```
+Once we save the files and because isFirstName is false, we will see that the name is no longer rendered on the browser. However, if we were to update this to be true, save that value so the condition resolves as true then the "John" first name will be rendered in the browser. If we set this first name back to false and we'll notice that the first name is no longer rendered in our browser, instead it's empty. That's the default behavior of the if statement - if the expression is true we render the provided template otherwise it's empty.
+If we were to achieve the same behavior with WPF, we would need to use a visibility converter. The code would look similar to the following:
+```cs
+public bool IsFirstName { get; set; }
+public Sample()
+  {
+      InitializeComponent();
+      this.DataContext = this;
+      this.IsFirstName = true;
+  }
+```
+```html
+<UserControl.Resources>
+  <BooleanToVisibilityConverter x:Key="VisibleIfTrueConverter" />
+</UserControl.Resources>
+<Grid>
+  <Label Visibility="{Binding Path=IsFirstName, Converter={StaticResource VisibleIfTrueConverter}}">John</Label>
+</Grid>
+```
+A lot easier and more straightforward it is in Angular.
+However, in this case, if "isFirstName" is false we want to provide a last name instead. To do that we're going to take advantage of the "else" clause of the `ngIf` directive. Let's start by creating an `ng-template` defining an "h2" tag which contains the last name. Now, an `ng-template` is simply a placeholder that allows us to define content that is not part of the DOM, but can be added via code such as using the `ngIf` directive. But, in order to use this in the directive, we need to give it a template reference variable name such as "lastname". Now that we have named our `ng-template`, let's go into our `ngIf` directive, add "; else lastname" and save this. Because "isFirstName" is false for saying else use the lastname, we are using the template with the last name.
```

Comment:
1. Maybe it is a good idea to bold either the property names or their values, for example:
... let's go into our ngIf directive, add **"; else lastname"** and save this
2. Because "isFirstName" is false for saying else use the lastname, we are using the template with the last name. **->** Because "isFirstName" is false and for saying **"else use the lastname"**, we are using the template with the last name.

Reply:
Addressed.